components/director/internal/systemfetcher/client.go

```diff
@@ -0,0 +1,125 @@
+package systemfetcher
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	urlpkg "net/url"
+	"time"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/log"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+var (
+	scopes = []string{"uaa.resource"}
+)
+
+type OAuth2Config struct {
+	ClientID                  string `envconfig:"APP_OAUTH_CLIENT_ID"`
+	ClientSecret              string `envconfig:"APP_OAUTH_CLIENT_SECRET"`
+	OAuthTokenEndpointPattern string `envconfig:"APP_OAUTH_TOKEN_ENDPOINT_PATTERN"`
+}
+
+type APIConfig struct {
+	Endpoint                    string        `envconfig:"APP_SYSTEM_INFORMATION_ENDPOINT"`
+	FilterCriteria              string        `envconfig:"APP_SYSTEM_INFORMATION_FILTER_CRITERIA"`
+	FilterTenantCriteriaPattern string        `envconfig:"APP_SYSTEM_INFORMATION_FILTER_TENANT_CRITERIA_PATTERN"`
+	Timeout                     time.Duration `envconfig:"APP_SYSTEM_INFORMATION_FETCH_TIMEOUT"`
+}
+
+type Client struct {
+	apiConfig         APIConfig
+	oAuth2Config      OAuth2Config
+	clientCreatorFunc ClientCreatorFunc
+}
+
+type ClientCreatorFunc func(ctx context.Context, oauth2Config OAuth2Config, scopes []string, tenant string) *http.Client
+
+func DefaultClientCreator(ctx context.Context, oAuth2Config OAuth2Config, scopes []string, tenant string) *http.Client {
```

Comment:
Unused `tenant` arg

Reply:
also no need to export func

Reply:
It needs to be exported, as it is used from main pkg. I think the tenant argument is used in tests to set different tenants

Reply:
I saw what you mean :D. Made the function type private.